userguide/tutorials/notification_plugin.adoc

```diff
@@ -1,31 +1,142 @@
+|SUBSCRIPTION_UNCANCEL
+|Occurs when a subscription cancellation is undone (Note that subscription cancellation  cannot be undone once the cancellation becomes effective )
+|SUBSCRIPTION_BCD_CHANGE
+|Occurs when the subscription billing date is changed
+|ENTITLEMENT_CREATION
+|Occurs when a new entitlement is created
+|ENTITLEMENT_CANCEL
+|Occurs when an entitlement is cancelled
+|BUNDLE_PAUSE
+|Occurs when a subscription bundle is paused
+|BUNDLE_RESUME
+|Occurs when a subscription bundle is resumed
+|OVERDUE_CHANGE
+|Occurs when there is a change is an overdue stage
+|INVOICE_CREATION
+|Occurs when a new invoice is generated
+|INVOICE_ADJUSTMENT
+|Occurs when an invoise is adjusted
+|INVOICE_NOTIFICATION
+|Occurs when an invoice is going to be generated for an account in the future (Can be used to notify customers about upcoming bills)
+|INVOICE_PAYMENT_SUCCESS
+|Occurs when an invoice payment is successful
+|INVOICE_PAYMENT_FAILED
+|Occurs when an invoice payment fails
+|PAYMENT_SUCCESS
+|Occurs when payment is successful
+|PAYMENT_FAILED
+|Occurs when payment fails
+|TAG_CREATION
+|Occurs when a new tag is associated with a Kill Bill resource (account, invoice, etc.)
+|TAG_DELETION
+|Occurs when a tag associated with a Kill Bill resource (account, invoice, etc.) is deleted
+|CUSTOM_FIELD_CREATION
+|Occurs when a custom field is created
+|CUSTOM_FIELD_DELETION
+|Occurs when a custom field is deleted
+|TENANT_CONFIG_CHANGE
+|Occurs when a Tenant configuration is modified (So, any changes to a catalog, overdue config, etc. within a tenant triggers this event)
+|TENANT_CONFIG_DELETION
+|Occurs when a Tenant configuration is deleted
+|===
+
+
+== Some common use cases for notification plugin
+
+This section lists some common use cases where you can use the notification plugin and also specifies how you can use the notification plugin in these scenarios.
+
+=== Doing something when an account is created/modified
+
+Sometimes, you may wish to perform some action like updating Salesforce when there is some account related activity. For this, you can write a custom notification plugin that handles the following events:
+
+. ACCOUNT_CREATION
 
-It is expected that the plugin will simply pass the same retry schedule for each retry iteration, but this is not enforced -- and left for the plugin to decide. Kill Bill will look at the most recent retry schedule attached to the exception currently being handled and determine what to do based on that. If for instance a first schedule included 2 retries 10 days apart, and then upon retrying one time, the new schedule now includes only 1 retry, the cycle of retries would end there (as the latest schedule contains only one retry and Kill Bill already retried one time).
+. ACCOUNT_CHANGE
 
-For example:
+You can obtain the account information using the following:
 
+`Account account = osgiKillbillAPI.getAccountUserApi().getAccountById(killbillEvent.getAccountId(), context)`
```

Comment:
It would be useful to describe what the `context` is. I'll send some pointers in Slack.

Reply:
I've updated the document, restructured it a bit to explain context